docs/api/ref/GetAgent.md

```diff
@@ -0,0 +1,253 @@
+
+
+
+## Persisted Connection Data
+
+The `getAgent()` function uses [`SessionStorage`](https://html.spec.whatwg.org/multipage/webstorage.html) ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage)) to persist information on an instance of an app and how it connected to a Desktop Agent in order to ensure a consistent connection type and instance Id, in case it navigates or is refreshed. Applications are not expected to interact with this information directly, rather it is set and used by the `getAgent()` implementation.
+
+The details persisted conform to the following type:
+
+```ts
+/** Type representing data on the Desktop Agent that an app 
+ *  connected to that is persisted by the getAgent function 
+ *  to be used when re-connecting (after a navigation or  
+ *  refresh event) and to ensure a consistent instanceId.  
+ */ 
+type DesktopAgentDetails = { 
+    /** The type of Desktop Agent connected to. Used to  
+     *  prevent an inadvertent switch to a different agent.*/ 
+    agentType: WebDesktopAgentType,
+
+    /** The URL that was previously sent to the Desktop Agent 
+     * to establish the app's identity.*/ 
+    identityUrl?: string,
+
+    /** The current URL at the time of the last connection to 
+     * a Desktop Agent.*/ 
+    actualUrl?: string,
+
+    /** Optional URL field that should be used to store any 
+     *  URL that was used to connect to a Desktop Agent. URLs 
+     *  may have been provided by a parent window that has since 
+     *  gone away and persisting may allow re-connection in such 
+     *  cases. */ 
+    agentUrl?: string,
+
+    /** The appId that was identified for the application by the 
+     * Desktop Agent.*/ 
+    appId: string,
+
+    /** The instanceId that was issued to the app by the Desktop  
+     * Agent. */ 
+    instanceId: string,
+
+    /** The instanceUuid that was issued to the app. This should be 
+     *  passed when connecting to the Desktop Agent to help  
+     *  identify that this app has connected before and which  
+     *  instance it is, enabling the Desktop Agent to reissue 
+     *  the same instanceId. The instanceUuid should never be shared 
+     *  with other applications and is not available through the 
+     *  FDC3 API, allowing it to be used as a shared secret with 
+     *  the Desktop Agent that issued the associated instanceId.*/ 
+    instanceUuid: string 
+} 
+ 
+/** Enumeration of values used to describe types of web-based 
+ *  Desktop Agent. Each 'type' refers to the means by which 
+ *  a connection to the agent is made and/or an interface to it 
+ *  received. */ 
+enum WebDesktopAgentType { 
+    /** Denotes Desktop Agents that inject the FDC3 interface  
+     *  at `window.fdc3`. */ 
+    PRELOAD = "PRELOAD",
```

Comment:
We agreed to store this data for the purposes of latching to the same type of connection as before - pretty sure it was something both you and Derek were keen on. It means that on reconnect you only have to attempt that one approach AND should only accept that one approach (e.g. go straight to failover - for faster reconnect), which will avoid an inadvertant switch to a different DA. Further, in the event that the parent has gone away, in the PROXY_URL case you can use the stored `agentUrl` URL to connect.

You might think that the presence of `agentUrl` is enough - but that doesn't differentiate the `PRELOAD`, `PROXY_PARENT`, `FAILOVER` cases. Finally,

Reply:
I'm very late to the party here, but yes I was indeed extremely keen on this.  In particular, in order to avoid the scenario where the app first connects to Agent X in the parent window, then the parent window is closed, then the app window is refreshed, and then the app uses the failover function to connect to Agent Y.  Which would be super-confusing for the user (it would likely seem non-deterministic to them, even though it would actually be deterministic.

Storing this data ensures that the app re-connects to the agent that originally opened it - or no agent at all.  And that it never connects to a different agent in the above scenario.

Reply:
I must admit, it's not clear to me why we're using SCREAMING_SNAKE_CASE for the members of this particular enum instead of PascalCase.

Perhaps there's a reason?

Reply:
Fixed in a0a43c4b0c177b832401c59851091a1d1aa99d5d